tokio/src/net/addr.rs

```diff
@@ -40,6 +40,61 @@ where
     }
 }
 
+cfg_dns! {
+    /// LookupHost contains resolved SocketAddrs.
+    ///
+    /// This type is created by the function [`lookup_host`]. See its documentation for more.
+    #[derive(Debug)]
+    pub struct LookupHost<T>
+    where
+        T: sealed::ToSocketAddrsPriv,
+        T: ToSocketAddrs<Future = sealed::MaybeReady>,
```

Comment:
Could you elaborate on why the bounds on `Future` and `Iter` are added?

Reply:
It's unable to compile otherwise. I'd be happy to reduce this to _just_ `sealed::MaybeReady` and `sealed::OneOrMore`, but I'm not sure of the proper incantation—trait bounds like these aren't my strong suit. I'd appreciate some pointers.